src/caikit_nlp_client/grpc_client.py

```diff
@@ -159,32 +188,45 @@ def _close(self):
     def __del__(self):
         self._close()
 
-    def _make_channel(self, host: str, port: int, **kwargs) -> grpc.Channel:
-        log.debug(f"Making a channel for {host}:{port} with these kwargs={kwargs}")
-        if host.strip() == "":
+    def _make_channel(
+        self,
+        host: str,
+        port: int,
+        *,
+        insecure: bool = False,
+        ca_cert: Optional[bytes] = None,
+        client_key: Optional[bytes] = None,
+        client_cert: Optional[bytes] = None,
+        server_cert: Optional[bytes] = None,
+    ) -> grpc.Channel:
+        if not host.strip():
             raise ValueError("A non empty host name is required")
-        if port <= 0:
-            raise ValueError("A non zero port is required")
+        if int(port) <= 0:
+            raise ValueError("A non zero port number is required")
 
         connection = f"{host}:{port}"
-        ca_cert = kwargs.get("ca_cert")
+        if insecure:
+            log.warning("Connecting over an insecure grpc channel")
+            return grpc.insecure_channel(connection)
+
+        credentials_kwargs: dict[str, bytes] = {}
         if ca_cert:
```

Comment:
With this logic (which can be simplified) this worked for me:

```python
        if ca_cert and not any((client_cert, client_key)):
            log.info("Connecting using provided CA certificate for secure channel")
            credentials_kwargs.update(root_certificates=ca_cert)
        elif client_key and client_cert and ca_cert:
            log.info("Connecting using mTLS for secure channel")
            # TODO: check if server_cert has the same meaning as ca_cert, and get
            # rid of it if so
            credentials_kwargs.update(
                root_certificates=ca_cert,
                private_key=client_key,
                certificate_chain=client_cert,
            )
        elif any((client_cert, client_key, ca_cert)):
            raise ValueError("mTLS requires client_cert, client_key and ca_cert")
```

Reply:
Done